cmd/frontend/backend/repos_vcs.go

```diff
@@ -57,7 +65,7 @@ func GitRepo(ctx context.Context, repo *types.Repo) (gitserver.Repo, error) {
 	return gitserver.Repo{Name: result.Repo.Name, URL: result.Repo.VCS.URL}, nil
 }
 
-func quickGitserverRepo(ctx context.Context, repo api.RepoName) (*gitserver.Repo, error) {
+func quickGitserverRepo(ctx context.Context, repo api.RepoName, serviceType *string) (*gitserver.Repo, error) {
```

Comment:
It seems like the code would be simpler if you used the empty string instead of nil.

```suggestion
func quickGitserverRepo(ctx context.Context, repo api.RepoName, serviceType string) (*gitserver.Repo, error) {
```